csharp/src/Apache.Arrow/Ipc/MessageSerializer.cs

```diff
@@ -57,16 +58,30 @@ internal static Schema GetSchema(Flatbuf.Schema schema)
             for (int i = 0; i < schema.FieldsLength; i++)
             {
                 Flatbuf.Field field = schema.Fields(i).GetValueOrDefault();
-
-                schemaBuilder.Field(
-                    new Field(field.Name, GetFieldArrowType(field), field.Nullable));
+                Field arrowField = FieldFromFlatbuffer(field);
+                schemaBuilder.Field(arrowField);
             }
 
             return schemaBuilder.Build();
         }
 
+        private static Field FieldFromFlatbuffer(Flatbuf.Field flatbufField)
+        {
+            System.Collections.Generic.List<Field> childFields = null;
+            if (flatbufField.ChildrenLength > 0)
+            {
+                childFields = new System.Collections.Generic.List<Field>();
```

Comment:
```suggestion
                childFields = new System.Collections.Generic.List<Field>(flatbufField.ChildrenLength);
```

Since we know up front how many we will need.

Actually, do we even need a `List` here? Could it simply be an array since we know how many we will need?

Reply:
Passing in an Array here would get converted to a `List` here anyway in the `StructType` constructor. I did make the change to specify the capacity upfront though

Reply:
It wouldn’t need to make a List in the constructor since the ctor now takes an IReadOnlyList. It could just hang on to that.

Reply:
Oh yea you're right. IReadOnlyList only needs `this[index]`. Fixed now.